### PR TITLE
Reconcile loop refactor - Rclone to use mover interface

### PR DESCRIPTION
**Pull request by tesshuflower**
**Describe what this PR does**
Refactors rclone reconcile to use the mover interface

**Is there anything that requires special attention?**
- LastSyncStartTime was added to status - to move to a common way of calculating LastSyncDuration
- e2e tests added to test multiple syncs with snapshot in replicationdestination
  - Restic had an issue with this scenario as multiple syncs with snapshot would not take a new snapshot.  Fix is included by ensuring the restic mover removes pvc snapshot annotations in the cleanup cycle.

**Related issues:**
https://github.com/backube/volsync/issues/18

### CODE REVIEW

File: test-kuttl/e2e/multi-sync-snapshot-rclone/10-sync-from-source.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+kind: ReplicationSource
+metadata:
+  name: source
+spec:
+  sourcePVC: data-source
+  trigger:
+    manual: firstsync
+  rclone:
+    rcloneConfigSection: "rclone-data-mover"
+    rcloneDestPath: "volsync-test-bucket2"
```
Comment (JohnStrunk) [inline test-kuttl/e2e/multi-sync-snapshot-rclone/10-sync-from-source.yaml:12]: We should get in the habit of naming the minio destinations based on the test so we don't end up w/ collisions across tests.

File: test-kuttl/e2e/multi-sync-snapshot-rclone/rclone.conf [hunk 2]
```diff
@@ -0,0 +1,8 @@
+[rclone-data-mover]
```
Comment (JohnStrunk) [inline test-kuttl/e2e/multi-sync-snapshot-rclone/rclone.conf:1]: This temp file shouldn't be checked in

File: controllers/mover/rclone/mover.go [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+	}
+	if dataPVC == nil || err != nil {
+		return mover.InProgress(), err
+	}
+
+	// Validate rCloneConfig Secret
+	rcloneConfigSecret, err := m.validateRcloneConfig(ctx)
+	if rcloneConfigSecret == nil || err != nil {
+		return mover.InProgress(), err
+	}
```
[the remark this answers is not in the document]
Comment (tesshuflower) [reply controllers/mover/rclone/mover.go:96]: Agreed, I do think it makes sense to do those validations upfront

File: controllers/rclone_mover_test.go [hunk 4]
```diff
@@ -276,6 +279,162 @@ var _ = Describe("ReplicationDestination [rclone]", func() {
...
+						// Now manually trigger another sync to generate another snapshot
+						//
+						manualTrigger := "testrightnow1"
+						Eventually(func() error {
+							// Put this in Eventually loop to avoid update issues (controller is also updating the rd)
+							err := k8sClient.Get(ctx, client.ObjectKeyFromObject(rd), rd)
+							if err != nil {
+								return err
+							}
+							time.Sleep(1 * time.Second)
```
[the remark this answers is not in the document]
Comment (tesshuflower) [reply controllers/rclone_mover_test.go:334]: oops, I may have done this while debugging a timing issue - will remove and make sure the tests still pass

File: test-kuttl/e2e/multi-sync-snapshot-rclone/10-sync-from-source.yaml (hunk 1 above)
Comment (JohnStrunk) [reply test-kuttl/e2e/multi-sync-snapshot-rclone/10-sync-from-source.yaml]: Thanks for fixing `simple-rclone` also!

**Review by JohnStrunk — APPROVED** [review]
/lgtm

**Pull request merged**